custom-exporter/exporter.py

```diff
@@ -44,6 +52,98 @@ def initialize(self):
+            try:
+                event_filter = event.create_filter(
+                    fromBlock=self.last_processed_block,
+                    argument_filters={address_param_name: node_address}
+                )
+                new_events.extend(event_filter.get_all_entries())
+            except Exception as e:
+                print(f"Error fetching {event.event_name} events: {str(e)}")
+
+        new_events.sort(key=lambda x: (x['blockNumber'], x['transactionIndex'], x['logIndex']))
+        self.events.extend(new_events)
+        self.events.sort(key=lambda x: (x['blockNumber'], x['transactionIndex'], x['logIndex']))
+
+        if new_events:
+            self.last_processed_block = new_events[-1]['blockNumber'] + 1
+
+    def calculate_uptime(self, node_address):
+        total_uptime = 0
+        current_start = None
+
+        for event in self.events:
+            event_type = event['event']
+            event_time = self.w3.eth.get_block(event['blockNumber'])['timestamp']
+
+            if event_type in ['NodeRegistered', 'NodeActivated']:
+                if current_start is None:
+                    current_start = event_time
+                self.last_status_change_time = event_time
+            elif event_type in ['NodeQuit', 'NodeSlashed']:
+                if current_start is not None:
+                    total_uptime += event_time - current_start
+                    current_start = None
+                self.last_status_change_time = event_time
+
+        if self.current_node_status:
+            total_uptime += int(time.time()) - self.last_status_change_time
+
+        self.uptime_gauge.labels(node_address=node_address).set(total_uptime)
+
+    def update_uptime(self, node_address):
+        self.fetch_events(node_address)
+        self.calculate_uptime(node_address)
+
+    def accumulate_uptime(self, node_address):
+        if self.current_node_status:
+            current_time = int(time.time())
+            accumulated_time = current_time - self.last_status_change_time
+            self.uptime_gauge.labels(node_address=node_address).inc(accumulated_time)
+            self.last_status_change_time = current_time
+
+    def update_aws_metrics(self, node_address):
+        urls = [
+            f"https://ynnnmyu5n1.execute-api.us-east-1.amazonaws.com/randcast-task-received?node_id={node_address}",
+            f"https://fxxk2yesr0.execute-api.us-east-1.amazonaws.com/randcast-listener-interrupted?node_id={node_address}",
+            f"https://3fxdughka1.execute-api.us-east-1.amazonaws.com/randcast-partial-signature-generation-time?node_id={node_address}"
+        ]
+
+        gauges = [self.task_received_gauge, self.listener_interrupted_gauge, self.partial_signature_generation_time_gauge]
+
+        for url, gauge in zip(urls, gauges):
+            result = self.fetch_data(url)
+            if result:
+                for item in result:
+                    value = item.get('sum') or item.get('average', 0)
```

Comment:
I don't quite understand here, what statistical items are required for each of the three metrics?

Reply:
I can get it now with 3 lambdas. Let's make it clearer here like the way for `event_definitions` above.